pandas/core/arrays/boolean.py

```diff
@@ -289,6 +289,13 @@ def _from_factorized(cls, values, original: "BooleanArray"):
     def _formatter(self, boxed=False):
         return str
 
+    @property
+    def _hasnans(self):
+        # Note: this is expensive right now! The hope is that we can
```

Comment:
Just out of curiosity in what way is this expensive? I would think should be a pretty quick op

Reply:
I suppose "relatively expensive" would be better. What I meant was that it's not a simple attribute lookup, since it makes two passes over the data.

If / when we have an optional mask, this will something like `self._mask is not None`.